### PR TITLE
svg objects

### PR DESCRIPTION
**Pull request by kaalleen**
This branch represents (without additional changes) the troubleshoot objects branch.

I found it easier to create to a new pull request than updating the old pull request since a lot has changed since then. It will make it easier to see the actual changes.

(Update) reference: #525

### CODE REVIEW

**Comment by kaalleen** [comment]
Accidentally I pushed this into here, actually I wanted to create a new branch for this. Sorry....

Last commit is related to #331

**Comment by lexelby** [comment]
No problem, easily fixed.  You can do something like this:

```
$ git checkout svg-objects
$ git reset --hard 49dde9a
$ git push -f
```

That will fix this pull request.  Then you can do this to make a new branch with the #331 fix:

```
$ git checkout -b kaalleen/fix-331
$ git cherry-pick 459cae2
$ git push
```

You sure have been busy today, thank you for going through all the old issues!  Thank you so much :)

**Comment by kaalleen** [comment]
> You sure have been busy today, thank you so much for going through all the old issues! :)

A bit housekeeping ;)
There are still so many issues out there... I kinda lost the overview, so I started to clean up a little.

File: lib/commands.py [hunk 1]
```diff
@@ -240,6 +240,14 @@ def is_command(node):
     return CONNECTION_START in node.attrib or CONNECTION_END in node.attrib
 
 
+def is_command_symbol(node):
+    symbol = None
+    xlink = node.get(XLINK_HREF, "")
+    if len(xlink) > 11:
```
Comment (lexelby) [inline lib/commands.py:246]: Maybe `xlink.startswith("#inkstitch_")` here so that it's more clear what you're doing?

File: lib/commands.py [hunk 2]
```diff
@@ -240,6 +240,14 @@ def is_command(node):
...
+        symbol = node.get(XLINK_HREF)[11:]
+    return symbol in OBJECT_COMMANDS or symbol in GLOBAL_COMMANDS
```
Comment (lexelby) [inline lib/commands.py:248]: Don't forget `LAYER_COMMANDS`. :)  `symbol in COMMANDS` should do the job.  `COMMANDS` should contain everything.

File: lib/elements/clone.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+from .fill import Fill
+from .polyline import Polyline
+from .satin_column import SatinColumn
+from .stroke import Stroke
+from .validation import ObjectTypeWarning, ValidationWarning
+
+
+class CloneWarning(ValidationWarning):
+    name = _("Clone Object")
+    description = _("There is one or more clone objects in this document.  "
```
Comment (lexelby) [inline lib/elements/clone.py:25]: "is" should be "are"

File: lib/elements/clone.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    steps_to_solve = [
+        _("If you want to convert the clone into a real element, follow these steps:"),
+        _("* Select the clone"),
+        _("* Run: Edit > Clone > Unlink Clone (Alt+Shift+D)")
+    ]
+
+
+class CloneSourceWarning(ObjectTypeWarning):
+    name = _("Clone is not embroiderable")
+    description = _("There is one ore more clone objects in this document. A clone must be a direct child of an embroiderable element. "
```
Comment (lexelby) [inline lib/elements/clone.py:36]: "is" -> "are" here too

**Pull request merged**